### PR TITLE
Make repository REUSE compliant and check licenses in CI

### LINKED ISSUE #1560: REUSE-ify repository
Spinoff of #752 - I will convert the repo to be REUSE (https://reuse.software/) compliant per internal discussion.

### PR DESCRIPTION
**Pull request by oddhack**
Closes #1560 

'reuse lint' works fine on the command line, but the CI script will take some tweaking. @HansKristian-Work is this repo using Appveyor instead of Github Actions? I edited the YML under .github/workflows/ and the job seems to be executing, but doesn't seem to be pulling a copy of the repo into the Docker contain reuse runs in, though I don't understand why immediately.

### CODE REVIEW

**Comment by oddhack** [comment]
> @HansKristian-Work is this repo using Appveyor instead of Github Actions?

Never mind, I discovered a mix of my problems and files ignored by .gitignore which shouldn't have been. All good now.

**Comment by oddhack** [comment]
@HansKristian-Work from the REUSE perspective this is ready to go, but CI is failing in the Windows step. Possibly I edited some Windows-specific file to add a copyright comment in an illegal way? It's hard to tell from the CI log - it just refers to a group of tests failing - and the deeper log it refers to doesn't seem to be getting saved as a CI artifact, so I'm kinda at a loss to diagnose this.

File: checkout_glslang_spirv_tools.sh [hunk 1]
```diff
@@ -1,8 +1,10 @@
 #!/bin/bash
+# Copyright 2016-2021 The Khronos Group Inc.
+# SPDX-License-Identifier: Apache-2.0
 
-GLSLANG_REV=3de5cfe50edecd001e6d703555284d9b10b3dd57
-SPIRV_TOOLS_REV=f3ccb633dfd7c5de1f9f0a2d2e9d7a25f2478206
-SPIRV_HEADERS_REV=f027d53ded7e230e008d37c8b47ede7cd308e19d
+GLSLANG_REV=4dcc12d1a441b29d5901bc708bb1343d29d6459f
```
Comment (HansKristian-Work) [inline checkout_glslang_spirv_tools.sh:5]: Why does this change? This causes the CI failure btw.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Nits.

**Comment by HansKristian-Work** [comment]
@oddhack Should I just go ahead and clean up the nits?

**Review by HansKristian-Work — APPROVED** [review]
Fixed the nits myself.

**Pull request merged**